### ISSUE TITLE
[Feature Request] Ability to preserve just generic arguments

### ISSUE THREAD

**Issue by namolnad** [nicklockwood/SwiftFormat#631]
For my company's codebase we'd like to be able to wrap before-first for all arguments, with the exception of generic arguments, which we'd like to preserve. I've played with some of the config options, but don't seem to be able to get this behavior. Perhaps there could be an enhancement to create a rule like: `--wrap-generics preserve`?

**Comment by nicklockwood** [comment]
Yeah, there isn't a separate option for this yet, but it makes sense to add one.

**Comment by drekka** [comment]
I think I'm in the same boat here. We have declarations like this:

```swift
open class ItemsView<ItemCell: CollectionCell<ItemView, ItemRow>,
                     Row: ItemsRow<ItemRow>,
                     ItemView: TapView<ItemRow>,
                     ItemRow: TapRowType>: View<Row>, UICollectionViewDelegate, UICollectionViewDataSource {
```

which isn't great but that's the code. Ideally I'd like it to format the same way. The rule being that any wrapped generic 
declarations are indented to the opening declaration

Alternatively I'd also like (if it's possible) for swift format to reconfigure the code to use `where` clauses instead of inheritance. So the above code would get changed to something like:

```swift
open class ItemsView<ItemCell, Row, ItemView, ItemRow>: View<Row>, UICollectionViewDelegate, UICollectionViewDataSource
    where ItemRow:TapRowType, 
          ItemView: TapView<ItemRow>, 
          Row: ItemsRow<ItemRow>, 
          ItemCell: CollectionCell<ItemView, ItemRow> {
```

But I'll assume that's a bit of an ask :-)
